### PR TITLE
chore: upgrade to cosmos-sdk v0.46.16

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/cosmos-sdk/issues/353

I cherry-picked the commits from v0.46.14 up to v0.46.16: https://github.com/cosmos/cosmos-sdk/compare/v0.46.14...v0.46.16

```shell
git cherry-pick 16d52690792e6e9446097ca5b48179a01a0c24e9^..7799bba7bc889288607576aa11655b5fc31a2da9
```

## Testing

Verified that I can still sign txs with from celestia-app using this cosmos-sdk

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,4 +1,6 @@
-go 1.19
+go 1.21
+
+toolchain go1.21.3
```
Comment (rootulp) [inline go.mod]: gahh my local go 1.21 auto bumped this when it shouldn't have... will revert

File: go.mod [hunk 2]
```diff
@@ -18,7 +18,7 @@ require (
 	github.com/cosmos/cosmos-sdk/db v1.0.0-beta.1
 	github.com/cosmos/go-bip39 v1.0.0
 	github.com/cosmos/iavl v0.19.6
-	github.com/cosmos/ledger-cosmos-go v0.13.2
+	github.com/cosmos/ledger-cosmos-go v0.12.4
```
Comment (rootulp) [inline go.mod:21]: ledger-cosmos-go release notes are sparse but 0.12.4 and 0.13.2 should contain the hotfix needed for Ledger to work on MacOS. 

Ref: https://github.com/cosmos/ledger-cosmos-go/releases

File: go.mod [hunk 3]
```diff
@@ -51,29 +51,29 @@ require (
 	github.com/tendermint/tendermint v0.34.28
 	github.com/tendermint/tm-db v0.6.7
 	github.com/tidwall/btree v1.5.0
-	golang.org/x/crypto v0.14.0
+	golang.org/x/crypto v0.7.0
```
Comment (rootulp) [inline go.mod:54]: hmm I don't think we should downgrade this. Nvm https://github.com/cosmos/cosmos-sdk/blob/7799bba7bc889288607576aa11655b5fc31a2da9/go.mod#L54

File: go.mod [hunk 4]
```diff
@@ -51,29 +51,29 @@ require (
...
 	golang.org/x/exp v0.0.0-20230310171629-522b1b587ee0
-	google.golang.org/genproto/googleapis/api v0.0.0-20230920204549-e6e6cdab5c13
-	google.golang.org/grpc v1.58.3
-	google.golang.org/protobuf v1.31.0
+	google.golang.org/genproto v0.0.0-20230306155012-7f2fa6fef1f4
+	google.golang.org/grpc v1.55.0
+	google.golang.org/protobuf v1.30.0
```
Comment (rootulp) [inline go.mod]: the only version number that diverges here is grpc https://github.com/cosmos/cosmos-sdk/blob/7799bba7bc889288607576aa11655b5fc31a2da9/go.mod#L56-L58

**Review by cmwaters — APPROVED** [review]

**Review by evan-forbes — APPROVED** [review]

**Pull request merged**